common/key_value_database/platform/lmdb/lmdb.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <iostream>
+#include <lmdb++.h>
+#include <cstdio>
+
+#include <common/common.hpp>
+#include <common/key_value_database_lmdb.hpp>
+
+namespace mender::common::key_value_database {
+
+class LmdbTransaction : public Transaction {
```

Comment:
Q: When declaring the class in the `cpp` file, why not fill in the functionality here?

Reply:
No strong reason, but easier to read maybe? It's a bit easier to see what is in the class, to get an overview, and then see the code later on. Also, we save one indentation level.